File: nos2jdbc-tutorial-spring/src/main/java/nos2jdbc/tutorial/spring/service/ClubService.java
package nos2jdbc.tutorial.spring.service;

import java.util.List;

import nos2jdbc.tutorial.spring.entity.Club;
import nos2jdbc.tutorial.spring.gen.service.ClubServiceBase;

import static nos2jdbc.tutorial.spring.gen.names.ClubNames.*;
import static org.seasar.extension.jdbc.operation.Operations.*;

import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional
public class ClubService extends ClubServiceBase {
    public List<Club> findAllWithMembers() {
	return select()
		.leftOuterJoin(clubMemberRelList())
		.leftOuterJoin(clubMemberRelList().member())
		.getResultList();
    }
    
    public Club getByOrder(int co) {
	return select().orderBy(asc(id())).offset(co).limit(1).getSingleResult();
    }

}

File: nos2jdbc-tutorial-kotlin-spring/src/main/kotlin/nos2jdbc/tutorial/kotlinspring/service/ClubService.kt
package nos2jdbc.tutorial.kotlinspring.service

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import nos2jdbc.tutorial.kotlinspring.entity.Club
import nos2jdbc.tutorial.kotlinspring.gen.service.ClubServiceBase

import nos2jdbc.tutorial.kotlinspring.gen.names.ClubNames.*
import org.seasar.extension.jdbc.operation.Operations.*


@Service
@Transactional
class ClubService(): ClubServiceBase() {
	
    public fun findAllWithMembers(): List<Club> {
	return select()
		.leftOuterJoin(clubMemberRelList())
		.leftOuterJoin(clubMemberRelList().member())
		.getResultList();
    }
    
    public fun getByOrderer(co: Int): Club {
	
	return select().orderBy(asc(id())).offset(co).limit(1).getSingleResult();
    }
}